scripts/data/gen-nodetime/src/relayer/lib/xrelayer.ts

```diff
@@ -0,0 +1,490 @@
+import os from "os";
+import yaml from "js-yaml";
+import fs from "fs";
+import { Bip39, Random } from "@cosmjs/crypto";
+import { DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
+import { GasPrice } from "@cosmjs/stargate";
+import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
+import { stringToPath } from "@cosmjs/crypto";
+import { Coin } from "@cosmjs/stargate";
+import { Link, IbcClient } from "@confio/relayer/build";
+import { orderFromJSON } from "@confio/relayer/build/codec/ibc/core/channel/v1/channel";
+export const ensureChainSetupMethod = "ensureChainSetup";
+export const createPathMethod = "createPath";
+export const getPathMethod = "getPath";
+export const listPathsMethod = "listPaths";
+export const getDefaultAccountMethod = "getDefaultAccount";
+export const getDefaultAccountBalanceMethod = "getDefaultAccountBalance";
+export const linkMethod = "link";
+export const startMethod = "start";
+
+type EnsureChainSetupResponse = {
+	// id is the chain id of chain.
+	id: string;
+};
+type Account = {
+	address: string;
+};
+type ChainConfig = {
+	chainId: string;
+	rpcAddr: string;
+	addrPrefix: string;
+	gasPrice: string;
+};
+type ConnectOptions = {
+	sourcePort: string;
+	sourceVersion: string;
+	targetPort: string;
+	targetVersion: string;
+	ordering: "ORDER_UNORDERED" | "ORDER_ORDERED";
+};
+type Connections = {
+	srcConnection: string;
+	destConnection: string;
+};
+type Endpoint = {
+	channelID?: string;
+	chainID: string;
+	portID: string;
+};
+type Path = {
+	id: string;
+	isLinked: boolean;
+	src: Endpoint;
+	dst: Endpoint;
+};
+type LinkResponse = {
+	linkedPaths: string[];
+	alreadyLinkedPaths: string[];
+};
+type StartResponse = {};
+type PacketHeights = {
+	packetHeightA: number;
+	packetHeightB: number;
+	ackHeightA: number;
+	ackHeightB: number;
+};
+type PathConfig = {
+	path: Path;
+	options?: ConnectOptions;
+	connections?: Connections;
+	relayerData?: PacketHeights;
+};
+type RelayerConfig = {
+	mnemonic: string;
+	chains?: Array<ChainConfig>;
+	paths?: Array<PathConfig>;
+};
+
+type LogMethod = (message: string) => Logger;
+
+interface Logger {
+	error: LogMethod;
+	warn: LogMethod;
+	info: LogMethod;
+	verbose: LogMethod;
+	debug: LogMethod;
+}
+
+class ConsoleLogger {
```

Comment:
this can be split to a seperate file.

Reply:
too small and only used here imo to warrant it but sure